src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -233,4 +237,112 @@ public NeighborNotifyEvent(World world, BlockPos pos, IBlockState state, EnumSet
             return notifiedSides;
         }
     }
+    
+    /**
+     * Fired before a world fires the random tick on the block, registred by the {@link BlockEvent.RandomBlockTickEventRegistry}. 
+     * <br>
+     * This event is called in WorldServer.updateBlocks
+     * <br>
+     * 
+     * If this event is canceled, the block won't do updates, but if you changed the IBlockState variable, the block in the world would be changed to the new one.
+     * <br>
+     * This event is called on the {@link MinecraftForge.#EVENT_BUS}
+     * <br>
+     * 
+     * This event is {@link Cancelable}
+     * 
+     * <br><br>
+     * <strong>!!!WARNING!!! If the block isn't registred by {@link RandomTickEvent.RandomTickEventRegistry}, the event won't be sent!!!</strong>
+     */
+    @Cancelable
+    public static class RandomTickEvent extends BlockEvent
+    {
+    	private IBlockState state;
+    	
+		public RandomTickEvent(World world, BlockPos pos, IBlockState state) {
+			super(world, pos, state);
```

Comment:
Not to mention the newline issue with those brackets. Some get newlines and others don't.